### PR TITLE
Add drop aware sender new type (fixes #13242) 

### LINKED ISSUE #13242: Add dropaware Sender new type
### Describe the feature

the way the rpc cache works is a single loop over actions:

https://github.com/paradigmxyz/reth/blob/3af2afe99528750dc7465433c4248fd1435bf3e0/crates/rpc/rpc-eth-types/src/cache/mod.rs#L345-L345

disk lookups also reuse that single channel and send actions back to that sender:

https://github.com/paradigmxyz/reth/blob/3af2afe99528750dc7465433c4248fd1435bf3e0/crates/rpc/rpc-eth-types/src/cache/mod.rs#L360-L360

in the meantime the oneshot response senders wait for that action:

https://github.com/paradigmxyz/reth/blob/3af2afe99528750dc7465433c4248fd1435bf3e0/crates/rpc/rpc-eth-types/src/cache/mod.rs#L222-L225

these response channels remain in the lru until the action arrives in the mainloop.

This is problematic if the spawned task panics because then nothing is emitted back to the mainloop.
for example if this operation panics
https://github.com/paradigmxyz/reth/blob/3af2afe99528750dc7465433c4248fd1435bf3e0/crates/rpc/rpc-eth-types/src/cache/mod.rs#L365-L372
no action is generated by that task.

we can solve this by introducing a drop aware sender wrapper that sends an action on drop, if not consumed:

like:

https://github.com/paradigmxyz/reth/blob/3af2afe99528750dc7465433c4248fd1435bf3e0/crates/rpc/rpc-eth-types/src/cache/mod.rs#L373-L376

```rust
struct ActionSender {
  blockhash: B256,
  tx: Option<UnboundedSender<...>
}

impl ActionSender {
  fn send_block(self,...)
}

impl Drop for ActionSender {
  if let Some(tx)
   ... send error
}
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ABD-AZE**
Fixes #13242

### CODE REVIEW

File: crates/rpc/rpc-eth-types/src/cache/mod.rs [hunk 1]
```diff
@@ -359,6 +403,8 @@ where
                                 let provider = this.provider.clone();
                                 let action_tx = this.action_tx.clone();
                                 let rate_limiter = this.rate_limiter.clone();
+                                let mut action_sender =
+                                    ActionSender { blockhash: block_hash, tx: Some(action_tx) };
```
Comment (mattsse) [inline crates/rpc/rpc-eth-types/src/cache/mod.rs:407]: I'd like a `fn new` for this

File: crates/rpc/rpc-eth-types/src/cache/mod.rs [hunk 2]
```diff
@@ -65,6 +65,50 @@ type HeaderLruCache<H, L> = MultiConsumerLruCache<B256, H, L, HeaderResponseSend
 pub struct EthStateCache<B: Block, R> {
     to_service: UnboundedSender<CacheAction<B, R>>,
 }
+/// Drop aware sender struct
+#[derive(Debug)]
+pub struct ActionSender<B: Block, R: Send + Sync> {
```
Comment (mattsse) [inline crates/rpc/rpc-eth-types/src/cache/mod.rs:70]: we can keep this private

```suggestion
struct ActionSender<B: Block, R: Send + Sync> {
```

**Review by mattsse — CHANGES_REQUESTED** [review]
cool, this is great.

only have a smol nit

**Review by mattsse — APPROVED** [review]

**Comment by ABD-AZE** [comment]
@mattsse should I label the **fn new** as const as directed by lint?

**Comment by mattsse** [comment]
yep,
and `cargo +nightly fmt`

**Comment by ABD-AZE** [comment]
@mattsse can you please enable the workflow for this PR.

**Review by mattsse — APPROVED** [review]

**Pull request merged**